### PR TITLE
merge release_23.1 into dev

### PR DESCRIPTION
**Pull request by martenson**
resolving conflict in `lib/galaxy/model/metadata.py`

please check @jdavcs @mvdbeek 

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jdavcs** [comment]
The merge is fine.

This is a good example of what may go wrong with code duplication and why I made the edit suggestion in #16931. Now we have two completely different statements: 
`session.query(galaxy.model.MetadataFile).filter_by(uuid=value).one_or_none()`
and
`session.execute(select(galaxy.model.MetadataFile).filter_by(uuid=value)).scalar_one_or_none()`
which do exactly the same thing. This makes that code rather unintuitive. I'll fix this.

**Comment by martenson** [comment]
You can push the refactoring here in the merge if you wish, so any confusion is short-lived.

File: lib/galaxy/model/metadata.py [hunk 1]
```diff
@@ -605,7 +605,16 @@ def wrap(self, value, session):
         if isinstance(value, int):
             return session.get(galaxy.model.MetadataFile, value)
         else:
-            return session.execute(select(galaxy.model.MetadataFile).filter_by(uuid=value)).scalar_one()
+            wrapped_value = session.query(galaxy.model.MetadataFile).filter_by(uuid=value).one_or_none()
```
Comment (jdavcs) [inline lib/galaxy/model/metadata.py]:
```suggestion
            wrapped_value = session.execute(select(galaxy.model.MetadataFile).filter_by(uuid=value)).scalar_one_or_none()
```
Let's keep the two statements identical (session.query... and session.execute...)

**Comment by jdavcs** [comment]
I don't think a merge PR requires approval, but just in case, +1, looks good. Thanks!

**Pull request merged**